Source/Dafny/Compilers/Compiler-Csharp.cs

```diff
@@ -1634,23 +1648,32 @@ protected class ClassWriter : IClassWriter {
     protected void DeclareField(string name, bool isPublic, bool isStatic, bool isConst, string typeName, string rhs, ClassWriter cw) {
       var publik = isPublic ? "public" : "private";
       var konst = isConst ? " readonly" : "";
+      var virtuall = DafnyOptions.O.Synthesize ? " virtual" : "";
       if (isStatic) {
         cw.StaticMemberWriter.Write($"{publik} static{konst} {typeName} {name}");
         if (rhs != null) {
           cw.StaticMemberWriter.Write($" = {rhs}");
         }
         cw.StaticMemberWriter.WriteLine(";");
-      } else if (cw.CtorBodyWriter == null) {
-        cw.InstanceMemberWriter.Write($"{publik}{konst} {typeName} {name}");
-        if (rhs != null) {
-          cw.InstanceMemberWriter.Write($" = {rhs}");
-        }
-        cw.InstanceMemberWriter.WriteLine(";");
       } else {
-        cw.InstanceMemberWriter.WriteLine($"{publik} {typeName} {name};");
-        if (rhs != null) {
-          cw.CtorBodyWriter.WriteLine($"this.{name} = {rhs};");
+        string ending = "";
+        if (isPublic) {
+          cw.InstanceMemberWriter.Write($"{publik} {virtuall} {typeName} {name} {{get; set;}}");
```

Comment:
It looks like you're ignoring `konst` here and emitting a setter even if `isConst` is true. Is that intentional?

Reply:
I borrowed this from @keyboardDrummer's proposed [solution](https://github.com/dafny-lang/dafny/pull/1809#discussion_r810911776), which does seem to emit the setter regardless. In my latest commit I have changed the code so that it only emits it if `isConst` is false, as below. Would this be the correct way of doing it?

```csharp
if (isPublic) {
  if (isConst) {
    cw.InstanceMemberWriter.Write($"{publik}{konst} {virtuall} {typeName} {name} {{get;}}");
  } else {
    cw.InstanceMemberWriter.Write($"{publik} {virtuall} {typeName} {name} {{get; set;}}");
   }
} else {
  cw.InstanceMemberWriter.WriteLine($"{publik}{konst} {typeName} {name}");
  ending = ";";
}
```